subsys/bluetooth/host/hci_core.c

```diff
@@ -1088,20 +1088,26 @@ static void update_pending_id(struct bt_keys *keys, void *data)
 }
 #endif
 
-static struct bt_conn *find_pending_connect(bt_addr_le_t *peer_addr)
+static struct bt_conn *find_pending_connect(u8_t role, bt_addr_le_t *peer_addr)
 {
 	struct bt_conn *conn;
 
 	/*
 	 * Make lookup to check if there's a connection object in
 	 * CONNECT or DIR_ADV state associated with passed peer LE address.
 	 */
-	conn = bt_conn_lookup_state_le(peer_addr, BT_CONN_CONNECT);
-	if (conn) {
+	if (IS_ENABLED(CONFIG_BT_CENTRAL) && role == BT_HCI_ROLE_MASTER) {
+		conn = bt_conn_lookup_state_le(peer_addr, BT_CONN_CONNECT);
 		return conn;
```

Comment:
This could be just `return bt_conn_lookup_state_le()`. Same for the other branch and then you can remove the conn variable completely from this helper function.

Reply:
It might look wierd when looking at this commit and not on the overall diff. I intentionally made this commit this way to avoid the next commit rewriting it.
Next commit adds:
```
		if (!conn) {
			conn = bt_conn_lookup_state_le(BT_ADDR_LE_NONE,
						       BT_CONN_CONNECT_ADV);
		}

```
Any objections to doing that? I'd prefer to keep the commits separate since one is refactor, the other a change.

Reply:
I missed that part. In that case it's perfectly fine.